### PR TITLE
Task#5 - Feat: group filter popover component

### LINKED ISSUE #52: Task#5 - DevTask: Create GroupFilterPopover component
## OVERVIEW
- Create GroupFilterPopover UI 
- Add hanlde to Close and Open Popover
## DEMO
image

### PR DESCRIPTION
**Pull request by TheHienDuong**
- Closes #52 
## OVERVIEW
**UI:**
- Create GroupFilterPopover
- Create Jsx.elemt popover open
   - UI - Menufacture list: Button brand filter
      - Using Over max-width, overflow-y-auto to process the button list into a responsive scroll
   - UI - Prices list: Button price filter 
- Update Popover
- Update comment
- Add arrow_filter icon below the filter
- Testing UI in App. jsx

**Handle**:
- Add useState to handle close and open popover ( boolean )

**Data**:
- Mock data in file json
- Create constant to save the image URL

## ISSUE ( RESOLVED )
- **_Issues_:**  The popover is overlapped when clicking the button to open the popover, but not closing the previous window when opening it

**The solution has already been mentioned by the mentor**

> - **Option1**: Adding one more event handle when the user clicks outside the popover container. React has one advanced feature called React hook and React's community has the solution for your issue called: [useClickOutside](https://www.30secondsofcode.org/react/s/use-click-outside/)
> - **Option2**: Implement "hover" then display the popover instead of click event

- Fix: Using **Option1** - [useClickOutside](https://www.30secondsofcode.org/react/s/use-click-outside/)
     - Apply useReft and useEffect 
- Why is choose **Option1**?
   - I'm following the click direction, and I also create an even handle for the button click, and in the popover, I also use the close button to close the button so my popover now uses click as well. Close and open can be used for both buttons inside
   - I thought about your option 2 but think it would be easier for me, but I'm in the direction of doing clicks, so I wanted to try it and see how the click side goes. So I choose option 1 
  
### UI

- **_GroupFilterPopover component _**

https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/113186169/1e3866d7-1962-4f8d-bede-8e0e7ed9a875

### CODE REVIEW

File: practice-two/src/components/GroupFilterPopover.jsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  const prices = [
+    '2-4 million',
+    '4-8 million',
+    '8-15 million',
+    'Over 15 million',
+  ];
+
+  return (
+    <div className="relative" ref={popoverRef}>
+      <Button onClick={handleButtonClick}>{children}</Button>
```
Comment (phattruong-agilityio) [inline practice-two/src/components/GroupFilterPopover.jsx]: what is this Button?
onClick={handleButtonClick} => meaning less
Why the Button receiver {children}. It's a specific component so IMO, you can do it normally without children

File: practice-two/src/components/GroupFilterPopover.jsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+const GroupFilterPopover = ({ children }) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const popoverRef = useRef(null);
+
+  // Toggle the popover open/close state
+  const togglePopover = () => {
+    setIsPopoverOpen(!isPopoverOpen);
+  };
+
+  // Handle button click event
```
Comment (phattruong-agilityio) [inline practice-two/src/components/GroupFilterPopover.jsx]: meaningless

File: practice-two/src/components/GroupFilterPopover.jsx (hunk 1 above)
Comment (TheHienDuong) [reply practice-two/src/components/GroupFilterPopover.jsx]: Updated, plz help me check

File: practice-two/src/components/GroupFilterPopover.jsx (hunk 2 above)
Comment (TheHienDuong) [reply practice-two/src/components/GroupFilterPopover.jsx]: Removed

File: practice-two/src/components/GroupFilterPopover.jsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    };
+  }, []);
+
+  // Data testing
+  const manufacturers = [
+    'http://cdn.tgdd.vn/Brand/1/logo-xiaomi-220x48-5.png',
+    // Add other manufacturer URLs here
+  ];
+
+  const prices = [
```
Comment (phattruong-agilityio) [inline practice-two/src/components/GroupFilterPopover.jsx]: Does this one should come from props?

File: practice-two/src/components/GroupFilterPopover.jsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    document.addEventListener('click', handleClickOutside);
+
+    return () => {
+      // Clean up the event listener when the component unmounts.
+      document.removeEventListener('click', handleClickOutside);
+    };
+  }, []);
+
+  // Data testing
+  const manufacturers = [
```
Comment (phattruong-agilityio) [inline practice-two/src/components/GroupFilterPopover.jsx]: Does this one should come from props?

File: practice-two/src/components/GroupFilterPopover.jsx (hunk 3 above)
Comment (TheHienDuong) [reply practice-two/src/components/GroupFilterPopover.jsx]: Done

File: practice-two/src/components/GroupFilterPopover.jsx (hunk 4 above)
Comment (TheHienDuong) [reply practice-two/src/components/GroupFilterPopover.jsx]: Done

**Pull request merged**